test/jdk/sun/security/provider/acvp/SHA_Test.java

```diff
@@ -0,0 +1,137 @@
+                            for (var r : c.get("resultsArray").asArray()) {
+                                if (alg.startsWith("SHA3-")) {
+                                    var MD = SEED;
+                                    for (var i = 0; i < 1000; i++) {
+                                        if (trunc) {
+                                            MD = Arrays.copyOf(MD, INITIAL_SEED_LENGTH / 8);
+                                        }
+                                        MD = md.digest(MD);
+                                    }
+                                    Asserts.assertEqualsByteArray(MD,
+                                            toByteArray(r.get("md").asString()));
+                                    SEED = MD;
+                                } else {
+                                    var A = SEED;
+                                    var B = SEED;
+                                    var C = SEED;
+                                    byte[] MD = null;
+                                    for (var i = 0; i < 1000; i++) {
+                                        var MSG = concat(A, B, C);
+                                        if (trunc) {
+                                            MSG = Arrays.copyOf(MSG, INITIAL_SEED_LENGTH / 8);
+                                        }
+                                        MD = md.digest(MSG);
+                                        A = B;
+                                        B = C;
+                                        C = MD;
+                                    }
+                                    Asserts.assertEqualsByteArray(MD,
+                                            toByteArray(r.get("md").asString()));
+                                    SEED = MD;
+                                }
+                            }
+                        } else {
+                            System.out.print("bits ");
+                        }
+                    }
+                }
+                case "LDT" -> {
+                    for (var c : t.get("tests").asArray()) {
+                        System.out.print(c.get("tcId").asString() + " ");
+                        var lm = c.get("largeMsg");
+                        var ct = toByteArray(lm.get("content").asString());
+                        var flen = Long.parseLong(lm.get("fullLength").asString());
+                        var clen = Long.parseLong(lm.get("contentLength").asString());
+                        var cc = 0L;
+                        while (cc < flen) {
+                            md.update(ct);
+                            cc += clen;
+                        }
+                        Asserts.assertEqualsByteArray(md.digest(),
+                                toByteArray(c.get("md").asString()));
+                    }
+                }
+                default -> throw new UnsupportedOperationException("Unknown testType");
```

Comment:
Including the unknown test type in the exception would speed debugging.

Reply:
Correct. Fixed.